Foundation/NSPathUtilities.swift

```diff
@@ -10,13 +10,20 @@
 import CoreFoundation
 
 public func NSTemporaryDirectory() -> String {
-    #if os(macOS) || os(iOS)
+#if os(Windows)
+    var cchLength: DWORD = GetTempPathW(0, nil)
+    var wszPath: UnsafeMutableBufferPointer = UnsafeMutableBufferPointer<WCHAR>.allocate(capacity: Int(cchLength + 1))
+    defer { wszPath.deallocate() }
+    _ = GetTempPathW(cchLength + 1, wszPath.baseAddress)
+    return String(decodingCString: wszPath.baseAddress!, as: UTF16.self)
+#else
+#if os(macOS) || os(iOS)
```

Comment:
This is absolutely where `canImport(Darwin)` should go, since the point of this block is t guard from using API that exists in that module.

Reply:
I still hate `canImport(Darwin)`, but fine.  Although, I think that it should be done in #1941 rather than in this change.  I'll update that in a little while.